libs/platform/auth/login/login.component.ts

```diff
@@ -41,37 +40,44 @@ export class AuthLoginComponent extends ContentMixin<LoginOptions>(LitElement) {
     new DefaultAuthLoginStrategy()
   );
 
-  protected doLogin$ = new Subject<LoginRequest>();
-
-  @subscribe()
-  protected login$ = this.doLogin$.pipe(
-    tap(() => {
-      this.isLoading = true;
-      this.hasError = false;
-    }),
-    switchMap((request) =>
-      this.authLoginStrategy.login(request).pipe(
-        catchError((e) => {
-          this.isLoading = false;
-          this.hasError = true;
-          return EMPTY;
-        })
-      )
-    ),
-    tap(() => (this.isLoading = false)),
-    switchMap(() => {
-      if (this.componentOptions?.disableRedirect) {
-        return EMPTY;
+  protected async doLogin(loginState: LoginRequest): Promise<void> {
```

Comment:
I wonder if we cannot do this in a simplified way:
- Store the state in a signal; update the state on login
- Use an ElementEffect to observe the state, and act accordingly
- Use a signal to observe whether the user is authenticated (`protected isAuthenticated = signal(this.authService.isAuthenticated());`)
- Use an ElementEffect to observe the authenticated state (redirect)

We've done something similar at the account component in checkout.

Reply:
> Store the state in a signal; update the state on login
But here state is a state of the form, we don't need to observe it actually.

> update the state on login
Here is the main issue, as long as we have a loginStrategy where we can have it's own redirect login.